cmd/kubeadm/app/cmd/join.go

```diff
@@ -159,17 +152,6 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 		Long:  joinLongDescription,
 		Run: func(cmd *cobra.Command, args []string) {
 
-			c, err := joinRunner.InitData(args)
-			kubeadmutil.CheckErr(err)
-
-			err = joinRunner.Run(args)
-			kubeadmutil.CheckErr(err)
-
-			// TODO: remove this once we have all phases in place.
-			// the method joinData.Run() itself should be removed too.
-			data := c.(*joinData)
-			err = data.Run()
-			kubeadmutil.CheckErr(err)
```

Comment:
This block should not go away completely, otherwise the top-level `kubeadm join` will do nothing. This part should be kept:

```
			c, err := joinRunner.InitData(args)
			kubeadmutil.CheckErr(err)

			err = joinRunner.Run(args)
			kubeadmutil.CheckErr(err)
```

Reply:
c declared and not used?

Reply:
I think in this case too it will throw error c declared and not used

Reply:
WDYT?

Reply:
I think this should look like:

```
err = joinRunner.Run(args)	
kubeadmutil.CheckErr(err)
```

since we don't use the `joinData` here, and `Run` will already call internally to `InitData`.

Reply:
And I think this is the reason tests are failing because the top-level `kubeadm join` command is not triggering any phases with the runner until this is called.

Reply:
test failing locally after these changes too

Reply:
the phase runner needs to handle running at this point, as @ereslibre pointed out.

Reply:
please only remove:
```
// TODO: remove this once we have all phases in place.
			// the method joinData.Run() itself should be removed too.
			data := c.(*joinData)
			err = data.Run()
			kubeadmutil.CheckErr(err)
```

Reply:
it says c declared but not used

Reply:
try:
```
_, err := joinRunner.InitData(args)
```

Reply:
No need to call to InitData explicitly, as Run will do it behind the covers, unless we want to print something here that uses the joinData.

Reply:
+1, forgot about that.
which means that we only need to initialize the runner and run it.

Reply:
still the test is failing locally just used 
err := joinRunner.Run(args)	
kubeadmutil.CheckErr(err)

Reply:
the tests seem to be failing because we are not handling flags correctly:
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/73452/pull-kubernetes-integration/44006/#k8siokubernetescmdkubeadmtestcmd-testcmdjoinargsmixeddiscovery-token_and_config

^ the above example allows mixing flags which shouldn't be allowed.

Reply:
solution for it?

Reply:
would you no be able to investigate the problem?

Reply:
ok i will investigate for it